modules/dynamodb/dynamodb_test.go

```diff
@@ -0,0 +1,258 @@
+	ctr2, err := tcdynamodb.Run(ctx, image2_2_1)
+	testcontainers.CleanupContainer(t, ctr2)
+	require.NoError(t, err)
+
+	// fetch client handle again
+	cli, err = getDynamoDBClient(ctr2)
+	require.NoError(t, err, "failed to get dynamodb client handle")
+
+	// list tables and verify
+
+	result, err := cli.ListTables(context.Background(), nil)
+	require.NoError(t, err, "dynamodb list tables operation failed")
+	require.Empty(t, result.TableNames, "table should not exist after restarting container")
+}
+
+func TestRun_shouldStartWithTelemetryDisabled(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := tcdynamodb.Run(ctx, image2_2_1, tcdynamodb.WithDisableTelemetry())
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+}
+
+func TestRun_shouldStartWithSharedDBEnabledAndTelemetryDisabled(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := tcdynamodb.Run(ctx, image2_2_1, tcdynamodb.WithSharedDB(), tcdynamodb.WithDisableTelemetry())
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+}
+
+func createTable(client *dynamodb.Client) error {
+	_, err := client.CreateTable(context.Background(), &dynamodb.CreateTableInput{
+		TableName: aws.String(tableName),
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String(pkColumnName),
+				KeyType:       types.KeyTypeHash,
+			},
+		},
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String(pkColumnName),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+		},
+		BillingMode: types.BillingModePayPerRequest,
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addDataToTable(client *dynamodb.Client, val string) error {
+	_, err := client.PutItem(context.Background(), &dynamodb.PutItemInput{
+		TableName: aws.String(tableName),
+		Item: map[string]types.AttributeValue{
+			pkColumnName: &types.AttributeValueMemberS{Value: val},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func queryItem(client *dynamodb.Client, val string) (string, error) {
+	output, err := client.GetItem(context.Background(), &dynamodb.GetItemInput{
+		TableName: aws.String(tableName),
+		Key: map[string]types.AttributeValue{
+			pkColumnName: &types.AttributeValueMemberS{Value: val},
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	result := output.Item[pkColumnName].(*types.AttributeValueMemberS)
+
+	return result.Value, nil
+}
+
+type dynamoDBResolver struct {
+	HostPort string
+}
+
+func (r *dynamoDBResolver) ResolveEndpoint(ctx context.Context, params dynamodb.EndpointParameters) (smithyendpoints.Endpoint, error) {
+	return smithyendpoints.Endpoint{
+		URI: url.URL{Host: r.HostPort, Scheme: "http"},
+	}, nil
+}
+
+// getDynamoDBClient returns a new DynamoDB client with the endpoint resolver set to the DynamoDB container's host and port
+func getDynamoDBClient(c *tcdynamodb.DynamoDBContainer) (*dynamodb.Client, error) {
```

Comment:
suggestion: this is returning an error only to check its not one, pass in t and use require to simplify.